roles/nsx/README.md

```diff
@@ -0,0 +1,162 @@
+# NSX-T
+
+Configure NSX-T components for the Master host.
+
+## Requirements
+
+* Ansible 2.4+
+
+Currently the nsx container docker image is not publically available.
+While we fix this you need the image nsx-ncp in a local private registry
+or do the following on all nodes before running the ansible installer.
+* `ansible-playbook [-i /path/to/inventory] playbooks/prerequisites.yml`
+* `docker load -i nsx-ncp-rhel-2.3.2.11627441.tar`
+* `docker image tag registry.local/2.3.2.11627441/nsx-ncp-rhel nsx-ncp`
+
+## Installation
+
+To install, set the following inventory configuration parameters:
+
+* `openshift_use_nsx=True`
+* `openshift_use_openshift_sdn=False`
+* `os_sdn_network_plugin_name='cni'`
+
+NSX specific parameters:
+
+* `nsx_openshift_cluster_name='ocp-cluster1'`
+(Required) NSX supports multiple Openshift/K8S clusters connected to the same 
+NSX Manager, so we mandate to provide cluster name in order to recognise.
+* `nsx_node_type='HOSTVM'`
+(Optional) Defaults to HOSTVM. Set to BAREMETAL if Openshift is not
+running in VMs.
+* `nsx_api_managers='10.10.10.10'`
+(Required) IP address of NSX Manager. If cluster of managers
+three ip addresses separated by comma.
+
+
+* `nsx_k8s_api_ip=192.168.10.10`
+(Optional) If set, NCP will talk to this ip, otherwise to kubernetes service ip.
+* `nsx_k8s_api_port=192.168.10.10`
+(Optional) Default to 443 for k8s service. Set to 8443 if you use it in combination
+with nsx_k8s_api_ip to specify master node ip.
+* `nsx_insecure_ssl=true`
+(Optional) Default is true as NSX Manager comes with untrusted certificate.
+If you have changed the certificate with a trusted one you can set it to false
+* `nsx_api_user='admin'`
+* `nsx_api_password='super_secret_password'`
+* `nsx_subnet_prefix=24`
+(Optional) Defaults to /24. This is the subnet size that will be dedicated per 
+Openshift Project. If number of PODs exceed the subnet size a new Logical Switch
+with the same subnet size will be added to the project.
+* `nsx_use_loadbalancer=true`
+(Optional) Defaults to true. Set to false if you don't want to use 
+NSX-T Loadbalancer for Route and Service type LoadBalancer
+* `nsx_lb_service_size='SMALL'`
+(Optional) Defaults to SMALL. Depending on the Edge size MEDIUM/LARGE 
+can also be possible
+* `nsx_tier0_router='MyT0Router'`
+(Required) Name or uuid of the precreated T0 router
```

Comment:
Same: its more important to know which params are required

Reply:
fixed with 11e4f05048c54c2d7b2d1f981cb101a0471be058